pkg/virt-controller/watch/vm.go

```diff
@@ -1546,9 +1548,80 @@ func (c *VMController) getControllerRevision(namespace string, name string) (*ap
 	return cr, nil
 }
 
-func (c *VMController) createVMRevision(vm *virtv1.VirtualMachine) (string, error) {
-	vmRevisionName := getVMRevisionName(vm.UID, vm.ObjectMeta.Generation)
-	createNotNeeded, err := c.deleteOlderVMRevision(vm)
+func (c *VMController) getVMSpecForKey(key string) (*virtv1.VirtualMachineSpec, error) {
+	obj, exists, err := c.crInformer.GetStore().GetByKey(key)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, fmt.Errorf("could not find key %s", key)
+	}
+
+	cr, ok := obj.(*appsv1.ControllerRevision)
+	if !ok {
+		return nil, fmt.Errorf("unexpected resource %+v", obj)
+	}
+
+	raw := map[string]interface{}{}
+	err = json.Unmarshal(cr.Data.Raw, &raw)
+	if err != nil {
+		return nil, err
+	}
+	patch, err := json.Marshal(raw["spec"])
+	if err != nil {
+		return nil, err
+	}
+	vmSpec := virtv1.VirtualMachineSpec{}
+	err = json.Unmarshal(patch, &vmSpec)
+	if err != nil {
+		return nil, err
+	}
+	return &vmSpec, nil
```

Comment:
I found there is `VirtualMachineRevisionData`, I think you can use it directly:

```
revisionData := VirtualMachineRevisionData{}
err = json.Unmarshal(cr.Data.Raw, &revisionData)
if err != nil {
    return nil, err
}
return revisionData.Spec
```

Reply:
Nice find, thanks!
Added as a separate commit.